yt_idv/scene_components/base_component.py

```diff
@@ -213,12 +217,14 @@ def run_program(self, scene):
                 self._set_uniforms(scene, p)
                 with self.data.vertex_array.bind(p):
                     self.draw(scene, p)
-        if self.cmap_min is None or self.cmap_max is None:
+        if self.cmap_min is None or self.cmap_max is None or self.last_use_db != self.use_db:
```

Comment:
can you remind me why you needed both `last_use_db` and `use_db`? I don't think you should change it here, I just suspect that it's a bandaid for some underlying issue in how we're refreshing/rendering.

Reply:
So every time we switch between using the depth/render buffer, we need to recompute the cmap_min/max values. But since it used to only run whenever the program was first started, I added the `or self.last_use_db != self.use_db` to detect if the state of the depth/render buffer selection changed.

And from what I've gathered, `run_program` is called a lot, so we shouldn't recompute these unless the state changes.

Reply:
ah right, the cmap. Maybe we should add a trigger for cmap recalculation to make this a bit clearer (not for this PR, I'll think about this more).